doc/user-manual/language/cubical.lagda.rst

```diff
@@ -726,8 +729,143 @@ is defined as:
 For many more examples of higher inductive types see:
 https://github.com/agda/cubical/tree/master/Cubical/HITs.
 
+Indexed inductive types
+=======================
+
+Cubical Agda has experimental support for the ``transp`` primitive when
+used to substitute the indices of an indexed inductive type. A handful
+of definitions (satisfying a technical restriction on their pattern
+matching) will compute when applied to a transport along indices. As an
+example of what works, let us consider the following running example:
+
+::
+
+  data Eq {a} {A : Set a} (x : A) : A → Set a where
+    reflEq : Eq x x
+
+Functions which match on ``Eq`` when all of its endpoints are variables,
+that is, very generic lemmas like ``symEq`` and ``transpEq`` below, will
+compute on all cases: they will compute to the given right-hand-side
+definitionally when their argument is ``reflEq``, and will compute to a
+transport in the codomain when their argument has been transported in
+the second variable.
+
+::
+
+  symEq : ∀ {a} {A : Set a} {x y : A} → Eq x y → Eq y x
+  symEq reflEq = reflEq
+
+  transpEq : ∀ {a} {A B : Set a} → Eq A B → A → B
+  transpEq reflEq x = x
+
+  pathToEq : ∀ {a} {A : Set a} {x y : A} → x ≡ y → Eq x y
+  pathToEq {x = x} p = transp (λ i → Eq x (p i)) i0 reflEq
+
+  module _ {a} {A B : Set a} {x y : A} {f : A ≃ B} where
+    _ : symEq (reflEq {x = x}) ≡ reflEq
+    _ = refl
+
+    _ : transpEq (pathToEq (ua (idEquiv Bool))) ≡ λ x → x
+    _ = refl
+
+The computational behaviour of matching on indexed inductives in very
+generic is suboptimal when compared to the behaviour of similar
+constructions implemented in terms of paths. For example, using the
+lemmas above, we can investigate ``uaβ`` for ``pathToEq``:
+
+::
+
+    uaβEq : transpEq (pathToEq (ua f)) ≡ f .fst
+    uaβEq = funExt λ z →
+      compPath (transportRefl (f .fst _))
+        (cong (f .fst) (compPath
+          (transportRefl _)
+          (compPath
+            (transportRefl _)
+            (transportRefl _))))
+
+In more concrete situations, such as when the indices are constructors
+of some other inductive type, pattern-matching definitions will not
+compute when applied to transports.
```

Comment:
Would this warrant an example?

Reply:
Ah I see this comes later, but is interrupted by the `NoEquivWhenSplitting` discourse.

Please add the usual forward referencing.